axelor-human-resource/src/main/java/com/axelor/apps/hr/service/project/ProjectTaskHRServiceImpl.java

```diff
@@ -0,0 +1,224 @@
+      AppProjectService appProjectService,
+      TaskStatusToolService taskStatusToolService,
+      TaskStatusProgressByCategoryRepository taskStatusProgressByCategoryRepository) {
+
+    super(
+        projectTaskRepo,
+        frequencyRepo,
+        frequencyService,
+        appBaseService,
+        projectRepository,
+        appProjectService,
+        taskStatusToolService,
+        taskStatusProgressByCategoryRepository);
+  }
+
+  @Override
+  public List<ProjectPlanningTime> updateProjectPlanningTime(
+      ProjectTask projectTask, ProjectTask projectTaskDb) {
+
+    List<ProjectPlanningTime> projectPlanningTimeList = projectTask.getProjectPlanningTimeList();
+
+    if (CollectionUtils.isEmpty(projectPlanningTimeList)) {
+      ProjectPlanningTime projectPlanningTime =
+          createProjectPlanningTime(projectTask, projectTask.getSprint());
+
+      if (projectPlanningTime != null) {
+        projectPlanningTimeList.add(projectPlanningTime);
+      }
+
+      return projectPlanningTimeList;
+    }
+
+    if (isInvalidInput(projectTask, projectTaskDb)) {
+      return projectPlanningTimeList;
+    }
+
+    Employee employee =
+        Optional.ofNullable(projectTask)
+            .map(ProjectTask::getAssignedTo)
+            .map(User::getEmployee)
+            .orElse(null);
+    Employee dbEmployee =
+        Optional.ofNullable(projectTaskDb)
+            .map(ProjectTask::getAssignedTo)
+            .map(User::getEmployee)
+            .orElse(null);
+    SprintPeriod sprintPeriod = projectTask.getSprint().getSprintPeriod();
+    SprintPeriod dbSprintPeriod = projectTaskDb.getSprint().getSprintPeriod();
+
+    LocalDateTime fromDateTime = sprintPeriod.getFromDate().atStartOfDay();
+    LocalDateTime toDateTime = sprintPeriod.getToDate().atStartOfDay();
+
+    boolean updated =
+        projectPlanningTimeList.stream()
+                .filter(
+                    planningTime ->
+                        shouldUpdatePlanningTime(planningTime, dbEmployee, dbSprintPeriod))
+                .peek(
+                    planningTime ->
+                        updateProjectPlanningTime(planningTime, employee, fromDateTime, toDateTime))
+                .count()
+            > 0;
+
+    if (!updated) {
+      ProjectPlanningTime projectPlanningTime =
+          createProjectPlanningTime(projectTask, projectTask.getSprint());
+
+      if (projectPlanningTime != null) {
+        projectPlanningTimeList.add(projectPlanningTime);
+      }
+    }
+
+    return projectPlanningTimeList;
+  }
+
+  private boolean isInvalidInput(ProjectTask projectTask, ProjectTask projectTaskDb) {
+
+    return projectTask.getSprint() == null
+        || projectTaskDb.getSprint() == null
+        || projectTask.getSprint().getSprintPeriod() == null
+        || projectTaskDb.getSprint().getSprintPeriod() == null
+        || projectTask.getAssignedTo().getEmployee() == null
+        || projectTaskDb.getAssignedTo().getEmployee() == null;
```

Comment:
Check only one projectTask in this function and call it twice to avoid code duplication